src/App.js

```diff
@@ -14,11 +15,108 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  }
+
+  addGoods = (good) => {
+    this.setState(prevState => ({
+      selectedGoods: [...prevState.selectedGoods, good],
+    }));
+  }
+
+  removeGood = (good) => {
+    this.setState(prevState => ({
+      selectedGoods: prevState.selectedGoods
+        .filter(product => product !== good),
+    }));
+  }
+
+  removeGoods = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  }
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div>
+        <div className="App">
+          {selectedGoods.length === 1
+            ? (
+              <h1>
+                {`${selectedGoods.join(', ')}
+                is selected`}
+              </h1>
+            )
+            : selectedGoods.length > 1
+              ? (
+                <h1>
+                  {`${selectedGoods.slice(0, -1).join(', ')}
+                and
+                ${selectedGoods[selectedGoods.length - 1]}
+                are selected`}
+                </h1>
+              )
+              : (<h1>No goods selected</h1>)
+          }
+
+          {selectedGoods.length !== 0
+          && (
+          <button
+            type="button"
+            className="btn btn-danger"
+            onClick={() => {
+              this.removeGoods();
+            }}
```

Comment:
якщо функція removeGoods не приймає параметрів то не потрібно створювати стрілочнку функцію
```suggestion
            onClick={this.removeGoods}
```